src/components/file-viewer/file-viewer.tsx

```diff
@@ -0,0 +1,225 @@
+import { Component, Element, h, Prop } from '@stencil/core';
+
+/**
+ * This is a smart component that detects most common file types such as
+ * image, audio, video, or text files,
+ * and renders them properly using native HTML5 elements.
+ *
+ * :::note
+ * Image files will always be contained in their containers, which means
+ * they automatically increases or decreases in size to fill the box
+ * whilst preserving thier aspect-ratio.
+ *
+ * Text and PDF files will also always respect the width and height of the
+ * container in which the `limel-file-viewer` is loaded.
+ * :::
+ *
+ * For some file types such as text and images, the component will display a
+ * download button and a button to open the file in a new browser tab.
+ * This will allow users to preview the file in a fullscreen more with the
+ * browser and take advantage of for example native zooming and panning
+ * functionalities.
+ *
+ *:::note
+ * The download button will not work in this documentation, since example files
+ * are not hosted in the same domain.
+ *:::
+ *
+ * @exampleComponent limel-example-file-viewer
+ * @exampleComponent limel-example-file-viewer-with-picker
+ */
+
+@Component({
+    tag: 'limel-file-viewer',
+    shadow: true,
+    styleUrl: 'file-viewer.scss',
+})
+export class FileViewer {
+    /**
+     * Link to the file
+     */
+    @Prop({ reflect: true })
+    public url: string;
+
+    /**
+     * Alternative text for assistive technologies and screen readers, displayed for image files
+     */
+    @Prop({ reflect: true })
+    public alt: string;
+
+    /**
+     * The file type
+     */
+    @Prop({ reflect: true })
+    public type: string;
+
+    /**
+     * Title of the button that downloads the file
+     */
+    @Prop({ reflect: true })
+    public titleDownload: string = 'Download';
+
+    /**
+     * Title of the button that opens the file in a new browser tab
+     */
+    @Prop({ reflect: true })
+    public titleOpenInNewTab: string = 'Open in a new tab';
+
+    /**
+     * Title of the button that opens the file for fullscreen viewing
+     */
+    @Prop({ reflect: true })
+    public titleOpenInFullscreen: string = 'Open in fullscreen';
+
+    /**
+     * Title of the button that exists the fullscreen mode
+     */
+    @Prop({ reflect: true })
+    public titleExitFullscreen: string = 'Exit fullscreen';
+
+    /**
+     * Message which will be displayed when the file type is not supported
+     */
+    @Prop({ reflect: true })
+    public messageNoFileTypeSupport: string = 'Cannot display this file!';
```

Comment:
We need to take a look at how translations are handled in general in Lime Elements, and I'm not sure this is the best approach to have a lot of props on the components. We have some other components that require translations as well, e.g. the date pickers and some other component that I have forgotten about right now :sweat_smile: 
For the date picker we have translation files in the `src/translations` folder. We should probably do it the same way here for now until we decide how to do it in general :slightly_smiling_face:

Reply:
That's a good point. We thought this was the way we handle them. But if there is a better way, let's do it ;)

Reply:
@jgroth resolved! [`9e61622` (#1153)](https://github.com/Lundalogik/lime-elements/pull/1153/commits/9e61622fb87c26630735d28758bcba43451b309c)